### ISSUE TITLE
setup:config:set --key append instead of replace

### ISSUE THREAD

**Issue by ju5t** [magento/magento2#11089]
### Preconditions
This particular container/server runs PHP 7.0.23 and Magento 2.1.9 which was downloaded from the website.

### Steps to reproduce

```
# bin/magento setup:config:set --key=test
Overwrite the existing configuration for key?[Y/n]Y
You saved the new configuration.
```

### Expected result

To replace the value in `crypt => array('key' => '')`.

### Actual result

It appends the value to `crypt => array('key' => '')`. This will result in:

```
  'crypt' =>
  array (
    'key' => 'test
test',
  )
```

**Comment by magento-engcom-team** [comment]
@ju5t, thank you for your report.
We've created internal ticket(s) MAGETWO-80169 to track progress on the issue.

**Comment by adioe3** [comment]
I'd like to take this one

**Comment by renttek** [comment]
Hi, there is already an "accepted" pull request from me for this issue. You can see the link right above your comment.
(I think, the existing pull requests are pretty bad to spot. Solved an Issue with existing PR last week too)

**Comment by magento-team** [comment]
Internal ticket to track issue progress: MAGETWO-80514

**Comment by okorshenko** [comment]
Hi @ju5t the issue has been fixed and delivered to 2.2-develop branch (#11155) and will be available with release 2.2.1

**Comment by ju5t** [comment]
Hi @okobchenko. Awesome work everyone. I'm impressed by the speed at which this was resolved.

**Comment by sidolov** [comment]
Hi @ju5t. Thank you for your report. 
The issue has been fixed in magento/magento2#13630 by @nmalevanec in 2.3-develop branch
Related commit(s):
 - [88a90f8b471958430022cbcc91c16d2a630fce97](https://github.com/magento/magento2/commit/88a90f8b471958430022cbcc91c16d2a630fce97)

The fix will be available with the upcoming 2.3.0 release.
